lib/senkyoshi/models/qti.rb

```diff
@@ -10,15 +10,16 @@
 }.freeze
 
 module Senkyoshi
-  class QTI < Resource
-    def self.from(data, pre_data)
+  class QTI < RootResource
+    def self.from(data, pre_data, _ = nil)
```

Comment:
That is the case here. This is so that all of the root resources have the same signature, and we don't have to care which kind of root resource we are calling `from` on. Adding `_ = nil` just makes it so we won't have to go update every test that calls `from` to take `nil` as a third argument.